cli/src/commands/metadata.rs

```diff
@@ -24,25 +25,45 @@ pub struct Opts {
     /// The format of the metadata to display: `json`, `hex` or `bytes`.
     #[clap(long, short, default_value = "bytes")]
     format: String,
+    /// Generate a subset of the metadata that contains only the
+    /// types needed to represent the provided pallets.
+    #[clap(long, use_value_delimiter = true, value_parser)]
+    retain_pallets: Option<Vec<String>>,
```

Comment:
I think I'd just call it `pallets`, ie "which pallets do I want".

It'd be amazing if this command supported a url _or_ filepath, so that somebody could point and strip down a metadata file they have already downloaded.

The `codegen` command already does, and it would be great if we could be consistent across those. This PR tried this and had a Source type to do this: https://github.com/paritytech/subxt/pull/805/files#diff-9e1e89a7d233bed7d98edc4c5e165f98d125c8d3e91f5417d278950858811039R9. I'd suggest similar; perhaps tweaking that to something like a:

```rust
enum MetadataSource {
    FilePath(PathBuf),
    Uri(Uri)
}

impl MetadataSource {
    pub async fetch(&self) -> RuntimeMetadataV14 { ... }
}
```

Reply:
Added in the latest commit, thanks!